liferay-content-management/liferay-portal
# LPD-19082 Index Asset Vocabulary Ids for assets

## Motivation

[Link to ticket](https://issues.liferay.com/browse/LPD-19082)

Index vocabularyIds just as we do with categories

## Notes:

* We cannot do an early return when there are no categories, because otherwise the property won't be set and searching will fail
* It's not behind a feature flag since it's only adding information to the index
* There will be a follow up PR with logic for searching the vocabularies and the related test

## Change summary:

* Refactor AssetCategoryDocumentContributor so that we call the database only once instead of 3 times when retrieving the vocabularies
* Add assetVocabularyIds and assetInternalVocabularyIds to the index
* New field for assetInternalVocabularyIds
* Adapt existing test

## Test Scenario 1

* Create a new site
* Create a new internal vocabulary and add a category to it
* Create a new public vocabulary and add a category to it
* Create a new web content and assign the categories to it
* Query the index, you will see the two new properties indexed

## Test Scenario 2

* Run the provided integration test
